docs/development.md

```diff
@@ -0,0 +1,110 @@
+
+[bok-choy]: https://github.com/edx/bok-choy
+
+Javascript tests exercise various javascript components of Group Project XBlock. They are written using [jasmine test
+framework][jasmine-test-framework] and are executed by [karma test runner][karma-test-runner]. Karma runs in node.js,
+so a compatible version of node.js and npm should be installed.
+
+Karma can run tests in two modes: continuous and single run. Single run mode executes all the tests once and than 
+terminates, presenting the results of run. In continuous mode, karma installs watchers on all the files included for the
+test run (both actual source files and files with tests) and re-runs the tests each time any file is updated.
+
+It is advised to use continuous mode for development. To start karma runner in development mode, issue the following
+
+    ./node_modules/.bin/karma start tests/js/karma.conf.js
+    
+Note that by default `karma-coverage` plugin is enabled. As a result, debugging JS tests might be problematic, as it
+modifies the source files on the fly by inserting coverage API calls. Simplest way to workaroud that is to comment 
+`coverage` reporter in `tests/js/karma.conf.js -> reporters` for the duration of debugging session (suggestions on how 
+to do that better are welcome). 
+
+[jasmine-test-framework]: http://jasmine.github.io/edge/introduction.html
+[karma-test-runner]: https://karma-runner.github.io/0.13/index.html
+
+## Code quality checks
+
+Code quality assertion tools are used to check both python (pep8 and pylint) and javascript (jshint) code quality.
+ 
+Both `group_project_v2` (actual code) and `tests` (tests, obviously) packages are checked. `pep8` is run with default 
+settings, except for `max-line-length=120`. `pylint` uses different pylintrc files for [group_project_v2 package]
+[gp-v2-pylintrc] and [tests package][tests-pylintrc].
+
+[gp-v2-pylintrc]: pylintrc
+[tests-pylintrc]: tests/pylintrc
+
+Javascript: same as with python code, both actual; code in `group_project_v2/public/js` and tests in `tests/js` are 
+checked, except for the vendor files in `group_project_v2/public/js/vendor`. Roughly equivalent jshintrc files are
+used for [actual code][gp-v2-jshintrc] and [tests][tests-jshintrc]
+
+[gp-v2-jshintrc]: .jshintrc
+[tests-jshintrc]: tests/.jshintrc
+
+[jshintrc]: .jshintrc
+
+## Continuous Integration build
+
+Travis CI build is configured to run on each PR against master. CI build installs Group Project XBlock v2 from scratch,
+runs unit, js and integration tests and checks code quality. CI build fails if there are any failing tests or quality
+code violations are reported.
+
+Sometimes CI build fails while tests run on development machine pass. To debug such problems, the first step is to 
+replicate CI build installation process and run as close as possible:
+
+* Commands run by CI are listed in `.travis.yml -> install` section.
```

Comment:
@e-kolpakov "run by CI" ⟶ "to prepare environment" (to make it consistent with the info in the next line)